packages/material-ui/src/Table/Table.js

```diff
@@ -23,6 +23,12 @@ export const styles = (theme) => ({
   stickyHeader: {
     borderCollapse: 'separate',
   },
+  /* Styles applied to the root element if `hideLastBorder={true}`. */
+  hideLastBorder: {
+    '& tbody tr:last-child th, & tbody tr:last-child td': {
```

Comment:
This breaks on custom components. It's currently allowed to render e.g. `div[role="rowgroup"]` instead of `tr`. Generally these selectors are too brittle and are hard to customize because you need to look at the source and then fight specificity.

Reply:
Yes you are right, we can pass the props to the TableContext and add style directly in TableCell.
What do you think?

Reply:
The point is that this selector can never be robust. The the points about override still stands.

Reply:
Yes I understand... mmmm... to make the solution more homogeneous we could add the props directly to the Cell component. 
In this case we don't have problem of overriding. If the user uses the Cell component he will decide to show border or not by using property and not CSS. 
In this call it will be only an alias of style={{border:0}} on the Cell.

I thought to pass the props throught context but then I don't know in the Cell when to apply the rule

Reply:
> you need to look at the source and then fight specificity

@eps1lon This is an excellent point. @marcosvega91 The lowest specificity selector I can come up with is:

```css
.MuiTable-hideLastBorder tr:last-child > *,
.MuiTable-hideLastBorder [role="row"]:last-child > *
```

which is "3" based on https://specificity.keegan.st/. Considering the prop is opt-in, maybe it's good enough?

Reply:
An alternative is a hideBorder prop at the `TableRow` level.

```css
.MuiTableRow-hideBorder:last-child > *
```

Reply:
At this point maybe it makes more sense to add  directly on row. It’ clearer I think

Reply:
Sorry, my selector was wrong, updated. There is also an option to rely on the context.